### PR TITLE
Add oauthaccesstokens to API_PREFERRED_VERSIONS, update deleteTokenLogoutService

### PR DESCRIPTION
**Pull request by benjaminapetersen**
- APIService in deleteTokenLogoutService would create a circular dependency

### CODE REVIEW

File: src/services/deleteTokenLogoutService.js [hunk 1]
```diff
@@ -6,7 +6,8 @@ angular.module('openshiftCommonServices')
 
   this.$get = function($q, $injector, Logger) {
     var authLogger = Logger.get("auth");
-
+    // Using APIService creates a circular dependency
```
Comment (spadgett) [inline src/services/deleteTokenLogoutService.js]: Does this work?

```js
var APIService = $injector.get("APIService");
```

Comment (benjaminapetersen) [reply src/services/deleteTokenLogoutService.js]: Yeah good call, like `DataService` below, updating

**Comment by benjaminapetersen** [comment]
updated

File: src/services/deleteTokenLogoutService.js [hunk 2]
```diff
@@ -19,11 +21,12 @@ angular.module('openshiftCommonServices')
 
         // Lazily get the data service. Can't explicitly depend on it or we get circular dependencies.
         var DataService = $injector.get('DataService');
+        var oAuthAccessTokensVersion = $injector.get('APIService').getPreferredVersion('oauthaccesstokens');
```
Comment (spadgett) [inline src/services/deleteTokenLogoutService.js]: Nit: I'd find this hard to read as one statement. I'd prefer something like...

```js
// Lazily get the API service and data service. Can't explicitly depend on them or we get circular dependencies.
var APIService = $injector.get('APIService');
var DataService = $injector.get('DataService');

var oAuthAccessTokensVersion = APIService.getPreferredVersion('oauthaccesstokens');
```

Comment (benjaminapetersen) [reply src/services/deleteTokenLogoutService.js]: I'm fine with that. updating.

**Comment by benjaminapetersen** [comment]
rebased

File: src/services/deleteTokenLogoutService.js [hunk 3]
```diff
@@ -18,12 +20,14 @@ angular.module('openshiftCommonServices')
         }
 
         // Lazily get the data service. Can't explicitly depend on it or we get circular dependencies.
```
Comment (spadgett) [inline src/services/deleteTokenLogoutService.js]: do you mind updating the comment since it applies to both services?

Comment (benjaminapetersen) [reply src/services/deleteTokenLogoutService.js]: Sure thing!

**Comment by benjaminapetersen** [comment]
updated

File: src/constants/apiPreferredVersions.js [hunk 4]
```diff
@@ -25,6 +25,7 @@ angular.module('openshiftCommonServices')
       imagestreamimages:                {group: 'image.openshift.io',         version: 'v1',      resource: 'imagestreamimages' },
       imagestreamimports:               {group: 'image.openshift.io',         version: 'v1',      resource: 'imagestreamimports' },
       limitranges:                      {version: 'v1',                       resource: 'limitranges' },
+      oauthaccesstokens:                {version: 'v1',                       resource: 'oauthaccesstokens' },
```
Comment (spadgett) [inline src/constants/apiPreferredVersions.js]: `group: oauth.openshift.io/v1`

Comment (benjaminapetersen) [reply src/constants/apiPreferredVersions.js]: ah, thx. got it.

**Comment by benjaminapetersen** [comment]
Think we are good on this one.

**Pull request merged**